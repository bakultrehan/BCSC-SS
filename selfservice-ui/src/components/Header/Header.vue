/** * Header of app */

<template>
  <div>
    <!-- <v-app-bar app> -->

    <v-app-bar app color class="header" light clipped-left>
      <v-app-bar-nav-icon
        @click.stop="drawer = !drawer"
        dark
      ></v-app-bar-nav-icon>
      <!-- <div class="banner"> -->
      <img
        src="@/assets/images/bc-logo-horizontal.svg"
        alt="Go to the Government of British Columbia website"
        class="img side-left-margin"
      />
      <!-- </a> -->
      <!-- <h1 class="text">BC SERVICE CARD</h1> -->
      <v-toolbar-title>BCSC SELF SERVICE</v-toolbar-title>
      <!-- </div> -->

      <v-spacer></v-spacer>
      <v-spacer></v-spacer>

      <v-btn
        text
        to="/about"
        link
        dark
        class="d-none d-sm-flex login-btn side-right-margin"
        v-if="!userProfile.firstName"
        >Login</v-btn
      >
      <v-toolbar-title v-if="userProfile.firstName"
        >Welcome {{ userProfile.firstName }} {{ userProfile.lastName }}

        <v-btn text @click="logout" color="white">
          <span class="mr-2" color="white">Logout</span>
        </v-btn>
      </v-toolbar-title>
      <v-btn text to="/" link dark class="mr-2 d-sm-none" @click="toggleMenu">
        <v-icon>mdi-menu</v-icon>
      </v-btn>
    </v-app-bar>
    <nav
      class="navigation-main"
      :class="{ 'active-menu': showMenu }"
      id="navbar"
      v-if="!showSideMenu"
    >
      <ul>
        <li class="d-sm-none">
          <v-btn text to="/about" link dark class="mr-2 login-btn">Login</v-btn>
        </li>
        <li>
          <router-link to="/">Home</router-link>
        </li>

        <li>
          <router-link to="about">About</router-link>
        </li>
      </ul>
    </nav>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
const KeyCloakModule = namespace('KeyCloakModule');

@Component
export default class Header extends Vue {
  @KeyCloakModule.Getter('userProfile') private userProfile!: [];
  @KeyCloakModule.Action('setLogout') private setLogout!: any;

  private showMenu: boolean = false;
  private drawer: boolean = false;
  private showSideMenu: boolean = false;
  /**
   * toggleMenu
   * @description : toggling mobile menu
   */
  private toggleMenu() {
    this.showMenu = !this.showMenu;
  }

  /**
   * make user logout
   * @returns {} dispatch
   */
  private logout() {
    this.setLogout();
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';

.header {
  background-color: $BCgovBlue5 !important;
  border-bottom: 2px solid $BCgovGold5;
  border-color: $BCgovGold5 !important;
  color: $BCgovWhite;
  top: 0px;
  width: 100%;

  .img {
    width: 175px;
  }
}

.v-toolbar__title {
  color: $BCgovWhite;
}

.navigation-main {
  display: none;
  position: fixed;
  top: 65px;
  color: $BCgovGold5;
  background-color: $BCgovBlue10;
  width: 100%;
  -webkit-box-shadow: 0 6px 8px -4px #b3b1b3;
  -moz-box-shadow: 0 6px 8px -4px #b3b1b3;
  box-shadow: 0 6px 8px -4px #b3b1b3;
  padding: 10px 0 10px 0;
  z-index: 1;
  //   margin-left: -16px;
  @include sm {
    display: block;
    top: 56px;
  }

  ul {
    display: flex;
    flex-direction: column;
    margin: 0;
    color: $BCgovWhite;
    list-style: none;
    @include sm {
      flex-direction: row;
      margin: 0 65px 0 65px;
    }
    li {
      margin: 5px 0;
      a {
        display: flex;
        font-size: 0.813em;
        font-weight: normal; /* 400 */
        color: $BCgovWhite;
        padding: 0 15px 0 15px;
        text-decoration: none;

        @include sm {
          border-right: 1px solid #9b9b9b;
        }
      }
    }
  }
}
.active-menu {
  display: block;
  top: 56px;
}

.navigation-main ul li a:hover {
  text-decoration: underline;
}

.navigation-main ul .router-link-exact-active {
  text-decoration: underline;
  font-weight: bold;
}

:focus {
  outline: 4px solid #3b99fc;
  outline-offset: 1px;
}
.login-btn {
  background-color: $BCgovGold5;
  color: $BCgovWhite;
  font-size: 18px;
}
</style>
