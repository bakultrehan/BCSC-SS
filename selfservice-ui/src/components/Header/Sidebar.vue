/** * Header of app */

<template>
  <v-card class="mx-auto">
    <v-navigation-drawer app clipped fixed height="100%" permanent>
      <!-- v-model="drawer" -->
      <v-list dark>
        <v-list-item>
          <v-list-item-content>
            <v-list-item-title class="title">Self Service</v-list-item-title>
            <!-- <v-list-item-subtitle>
              subtext
            </v-list-item-subtitle>-->
          </v-list-item-content>
        </v-list-item>

        <v-divider></v-divider>

        <v-list dense nav>
          <v-list-item v-for="item in items" :key="item.title" link>
            <v-list-item-icon>
              <v-icon>{{ item.icon }}</v-icon>
            </v-list-item-icon>

            <v-list-item-content>
              <v-list-item-title :to="item.title">
                <Link :to="item.link">{{ item.title }}</Link>
              </v-list-item-title>
            </v-list-item-content>
          </v-list-item>
        </v-list>
      </v-list>
    </v-navigation-drawer>
  </v-card>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import Link from '@/Atomic/Link/Link.vue';

const KeyCloakModule = namespace('KeyCloakModule');

@Component({
  components: {
    Link
  }
})
export default class Sidebar extends Vue {
  @Prop() private drawer!: boolean;
  private items: any = [
    { title: 'Dashboard', icon: 'mdi-view-dashboard', link: '/' },
    { title: 'Create Project', icon: 'mdi-plus-circle', link: '/project/info' },
    { title: 'About', icon: 'mdi-help-box', link: '/about' }
  ];
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';

.v-navigation-drawer {
  text-align: left;
  background-color: $BCgovBlue5 !important;
}
</style>
