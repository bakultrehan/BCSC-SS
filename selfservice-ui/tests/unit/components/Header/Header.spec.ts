/* tslint:disable */
import { mount, createLocalVue } from '@vue/test-utils';
import Header from '@/components/Header/Header.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import VueRouter from 'vue-router';

describe('Header.vue', () => {
  let wrapper: any;

  let vuetify: any;
  const localVue = createLocalVue();

  localVue.use(Vuetify);
  vuetify = new Vuetify();
  localVue.use(VueRouter);
  const router = new VueRouter();
  localVue.use(Vuex);
  const store = new Vuex.Store({
    modules: {
      KeyCloakModule: {
        namespaced: true,
        state: {},
        getters: {
          userProfile: jest.fn(() => {
            return {
              firstName: 'user',
              lastName: 'lastName'
            };
          }),
          isLoggedin: jest.fn(() => {
            return true;
          })
        }
      }
    }
  });
  const mountFunction = (options: any) => {
    return mount(Header, {
      vuetify,
      localVue,
      router,
      store,
      ...options
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders props when passed', () => {
    wrapper = mountFunction({});
    expect(wrapper.element).toMatchSnapshot();
  });
  it('Toggle menu onlick ', () => {
    const wrapper = mountFunction({});
    const event = jest.fn();
    const button = wrapper.find('.toggleMenu');
    wrapper.vm.$on('action-btn:clicked', event);
    button.trigger('click');

    expect(wrapper.element).toMatchSnapshot();
  });

  it('Logout user on logout button onlick ', () => {
    const wrapper = mountFunction({});

    const event = jest.fn();
    const button = wrapper.find('.logout');
    wrapper.vm.$on('action-btn:clicked', event);
    button.trigger('click');

    expect(wrapper.element).toMatchSnapshot();
  });
});
