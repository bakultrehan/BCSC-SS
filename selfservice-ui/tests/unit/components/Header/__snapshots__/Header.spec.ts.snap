// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue Logout user on logout button onlick  1`] = `
<div>
  <header
    class="header v-sheet v-sheet--tile theme--light v-toolbar v-app-bar v-app-bar--clipped v-app-bar--fixed"
    style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <img
        alt="Go to the Government of British Columbia website"
        class="img side-left-margin"
        src="@/assets/images/bc-logo-horizontal.svg"
      />
       
      <div
        class="v-toolbar__title"
      >
        BCSC SELF SERVICE
      </div>
       
      <div
        class="spacer"
      />
       
      <div
        class="spacer"
      />
       
      <!---->
       
      <div
        class="v-toolbar__title"
      >
        
      Welcome user lastName
      
        <button
          class="v-btn v-btn--flat v-btn--text theme--light v-size--default white--text"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            <span
              class="mr-2 logout"
              color="white"
            >
              Logout
            </span>
          </span>
        </button>
      </div>
       
      <a
        class="mr-2 d-sm-none toggleMenu v-btn--active v-btn v-btn--flat v-btn--router v-btn--text theme--dark v-size--default"
        href="#/"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-menu theme--dark"
          />
        </span>
      </a>
    </div>
  </header>
   
  <!---->
   
  <div
    class="mx-auto v-card v-sheet theme--light"
  >
    <nav
      class="v-navigation-drawer v-navigation-drawer--clipped v-navigation-drawer--fixed v-navigation-drawer--open theme--light"
      style="height: 100%; top: 64px; max-height: calc(100% - 64px); transform: translateX(0%); width: 256px;"
    >
      <div
        class="v-navigation-drawer__content"
      >
        <div
          class="v-list v-sheet v-sheet--tile theme--dark"
        >
          <div
            class="v-list-item theme--dark"
            tabindex="-1"
          >
            <div
              class="v-list-item__content"
            >
              <div
                class="v-list-item__title title"
              >
                
            Self Service
          
              </div>
            </div>
          </div>
           
          <hr
            aria-orientation="horizontal"
            class="v-divider theme--dark"
            role="separator"
          />
           
          <div
            class="v-list v-sheet v-sheet--tile theme--dark v-list--dense v-list--nav"
          >
            <div
              class="v-list-item v-list-item--link theme--dark"
              tabindex="0"
            >
              <div
                class="v-list-item__icon"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-view-dashboard theme--dark"
                />
              </div>
               
              <div
                class="v-list-item__content"
              >
                <div
                  class="v-list-item__title"
                  to="Dashboard"
                >
                  <a
                    class="router-link-main router-link-exact-active router-link-active"
                    href="#/"
                  >
                    
                Dashboard
              
                  </a>
                </div>
              </div>
            </div>
            <div
              class="v-list-item v-list-item--link theme--dark"
              tabindex="0"
            >
              <div
                class="v-list-item__icon"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-plus-circle theme--dark"
                />
              </div>
               
              <div
                class="v-list-item__content"
              >
                <div
                  class="v-list-item__title"
                  to="Create Project"
                >
                  <a
                    class="router-link-main"
                    href="#/project"
                  >
                    
                Create Project
              
                  </a>
                </div>
              </div>
            </div>
            <div
              class="v-list-item v-list-item--link theme--dark"
              tabindex="0"
            >
              <div
                class="v-list-item__icon"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-help-box theme--dark"
                />
              </div>
               
              <div
                class="v-list-item__content"
              >
                <div
                  class="v-list-item__title"
                  to="About"
                >
                  <a
                    class="router-link-main"
                    href="#/about"
                  >
                    
                About
              
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="v-navigation-drawer__border"
      />
    </nav>
  </div>
</div>
`;

exports[`Header.vue Toggle menu onlick  1`] = `
<div>
  <header
    class="header v-sheet v-sheet--tile theme--light v-toolbar v-app-bar v-app-bar--clipped v-app-bar--fixed"
    style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <img
        alt="Go to the Government of British Columbia website"
        class="img side-left-margin"
        src="@/assets/images/bc-logo-horizontal.svg"
      />
       
      <div
        class="v-toolbar__title"
      >
        BCSC SELF SERVICE
      </div>
       
      <div
        class="spacer"
      />
       
      <div
        class="spacer"
      />
       
      <!---->
       
      <div
        class="v-toolbar__title"
      >
        
      Welcome user lastName
      
        <button
          class="v-btn v-btn--flat v-btn--text theme--light v-size--default white--text"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            <span
              class="mr-2 logout"
              color="white"
            >
              Logout
            </span>
          </span>
        </button>
      </div>
       
      <a
        class="mr-2 d-sm-none toggleMenu v-btn--active v-btn v-btn--flat v-btn--router v-btn--text theme--dark v-size--default"
        href="#/"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-menu theme--dark"
          />
        </span>
      </a>
    </div>
  </header>
   
  <!---->
   
  <div
    class="mx-auto v-card v-sheet theme--light"
  >
    <nav
      class="v-navigation-drawer v-navigation-drawer--clipped v-navigation-drawer--fixed v-navigation-drawer--open theme--light"
      style="height: 100%; top: 64px; max-height: calc(100% - 64px); transform: translateX(0%); width: 256px;"
    >
      <div
        class="v-navigation-drawer__content"
      >
        <div
          class="v-list v-sheet v-sheet--tile theme--dark"
        >
          <div
            class="v-list-item theme--dark"
            tabindex="-1"
          >
            <div
              class="v-list-item__content"
            >
              <div
                class="v-list-item__title title"
              >
                
            Self Service
          
              </div>
            </div>
          </div>
           
          <hr
            aria-orientation="horizontal"
            class="v-divider theme--dark"
            role="separator"
          />
           
          <div
            class="v-list v-sheet v-sheet--tile theme--dark v-list--dense v-list--nav"
          >
            <div
              class="v-list-item v-list-item--link theme--dark"
              tabindex="0"
            >
              <div
                class="v-list-item__icon"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-view-dashboard theme--dark"
                />
              </div>
               
              <div
                class="v-list-item__content"
              >
                <div
                  class="v-list-item__title"
                  to="Dashboard"
                >
                  <a
                    class="router-link-main router-link-exact-active router-link-active"
                    href="#/"
                  >
                    
                Dashboard
              
                  </a>
                </div>
              </div>
            </div>
            <div
              class="v-list-item v-list-item--link theme--dark"
              tabindex="0"
            >
              <div
                class="v-list-item__icon"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-plus-circle theme--dark"
                />
              </div>
               
              <div
                class="v-list-item__content"
              >
                <div
                  class="v-list-item__title"
                  to="Create Project"
                >
                  <a
                    class="router-link-main"
                    href="#/project"
                  >
                    
                Create Project
              
                  </a>
                </div>
              </div>
            </div>
            <div
              class="v-list-item v-list-item--link theme--dark"
              tabindex="0"
            >
              <div
                class="v-list-item__icon"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-help-box theme--dark"
                />
              </div>
               
              <div
                class="v-list-item__content"
              >
                <div
                  class="v-list-item__title"
                  to="About"
                >
                  <a
                    class="router-link-main"
                    href="#/about"
                  >
                    
                About
              
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="v-navigation-drawer__border"
      />
    </nav>
  </div>
</div>
`;

exports[`Header.vue renders props when passed 1`] = `
<div>
  <header
    class="header v-sheet v-sheet--tile theme--light v-toolbar v-app-bar v-app-bar--clipped v-app-bar--fixed"
    style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <img
        alt="Go to the Government of British Columbia website"
        class="img side-left-margin"
        src="@/assets/images/bc-logo-horizontal.svg"
      />
       
      <div
        class="v-toolbar__title"
      >
        BCSC SELF SERVICE
      </div>
       
      <div
        class="spacer"
      />
       
      <div
        class="spacer"
      />
       
      <!---->
       
      <div
        class="v-toolbar__title"
      >
        
      Welcome user lastName
      
        <button
          class="v-btn v-btn--flat v-btn--text theme--light v-size--default white--text"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            <span
              class="mr-2 logout"
              color="white"
            >
              Logout
            </span>
          </span>
        </button>
      </div>
       
      <a
        class="mr-2 d-sm-none toggleMenu v-btn--active v-btn v-btn--flat v-btn--router v-btn--text theme--dark v-size--default"
        href="#/"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-menu theme--dark"
          />
        </span>
      </a>
    </div>
  </header>
   
  <!---->
   
  <div
    class="mx-auto v-card v-sheet theme--light"
  >
    <nav
      class="v-navigation-drawer v-navigation-drawer--clipped v-navigation-drawer--fixed v-navigation-drawer--open theme--light"
      style="height: 100%; top: 64px; max-height: calc(100% - 64px); transform: translateX(0%); width: 256px;"
    >
      <div
        class="v-navigation-drawer__content"
      >
        <div
          class="v-list v-sheet v-sheet--tile theme--dark"
        >
          <div
            class="v-list-item theme--dark"
            tabindex="-1"
          >
            <div
              class="v-list-item__content"
            >
              <div
                class="v-list-item__title title"
              >
                
            Self Service
          
              </div>
            </div>
          </div>
           
          <hr
            aria-orientation="horizontal"
            class="v-divider theme--dark"
            role="separator"
          />
           
          <div
            class="v-list v-sheet v-sheet--tile theme--dark v-list--dense v-list--nav"
          >
            <div
              class="v-list-item v-list-item--link theme--dark"
              tabindex="0"
            >
              <div
                class="v-list-item__icon"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-view-dashboard theme--dark"
                />
              </div>
               
              <div
                class="v-list-item__content"
              >
                <div
                  class="v-list-item__title"
                  to="Dashboard"
                >
                  <a
                    class="router-link-main router-link-exact-active router-link-active"
                    href="#/"
                  >
                    
                Dashboard
              
                  </a>
                </div>
              </div>
            </div>
            <div
              class="v-list-item v-list-item--link theme--dark"
              tabindex="0"
            >
              <div
                class="v-list-item__icon"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-plus-circle theme--dark"
                />
              </div>
               
              <div
                class="v-list-item__content"
              >
                <div
                  class="v-list-item__title"
                  to="Create Project"
                >
                  <a
                    class="router-link-main"
                    href="#/project"
                  >
                    
                Create Project
              
                  </a>
                </div>
              </div>
            </div>
            <div
              class="v-list-item v-list-item--link theme--dark"
              tabindex="0"
            >
              <div
                class="v-list-item__icon"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-help-box theme--dark"
                />
              </div>
               
              <div
                class="v-list-item__content"
              >
                <div
                  class="v-list-item__title"
                  to="About"
                >
                  <a
                    class="router-link-main"
                    href="#/about"
                  >
                    
                About
              
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="v-navigation-drawer__border"
      />
    </nav>
  </div>
</div>
`;
